source/device/device.hpp

```diff
@@ -18,7 +18,7 @@ class IProcessingDevice {
     // One step of device work cycle;
     // e.g. see next inlink, take one packet from it,
     // and do smth with it (send further, send ask etc)
-    virtual Time process() = 0;
+    virtual Time process(Time current_time) = 0;
```

Comment:
I think we do not need it as now we have `Simulator::get_current_time()`
```suggestion
    virtual Time process() = 0;
```